### ISSUE TITLE
[Code to raml] Controller inheritance is ignored

### ISSUE THREAD

**Issue by bendem** [phoenixnap/springmvc-raml-plugin#98]
~~Only the methods from the concrete class are found, all annotated methods on parent classes are ignored.~~

It seems that all the methods are found but the `@RequestMapping` on the child class is ignored so all inherited methods point to the same path and override each other.

**Comment by kurtpa** [comment]
Just to confirm that i'm understanding

Assuming Class Child extends Parent

class Parent {
       
       @RequestMapping("/something")
       public String doSomething () {
       }
}

@RequestMapping("/child")
class Child  extends Parent {
}

Is this the case?

**Comment by kurtpa** [comment]
so if i'm understanding well, doSomething is being mapped to /something, when it should be /child/something ?

**Comment by bendem** [comment]
Yes

**Comment by bendem** [comment]
Here is a sample fix

```diff
commit ad8f8855bff7fcb346beee959ebbf600f3095a9e
Author: Benjamin Demarteau <benjamin.demarteau@yelido.be>
Date:   Fri Oct 21 15:23:13 2016 +0200

    Fix retrieving resource path in case of inherited methods

diff --git a/springmvc-raml-parser/src/main/java/com/phoenixnap/oss/ramlapisync/parser/ResourceParser.java b/springmvc-raml-parser/src/main/java/com/phoenixnap/oss/ramlapisync/parser/ResourceParser.java
index bf60f40..68bf0f4 100644
--- a/springmvc-raml-parser/src/main/java/com/phoenixnap/oss/ramlapisync/parser/ResourceParser.java
+++ b/springmvc-raml-parser/src/main/java/com/phoenixnap/oss/ramlapisync/parser/ResourceParser.java
@@ -83,7 +83,7 @@ public abstract class ResourceParser {
 		try {
 			for (Method method : clazz.getMethods()) {
 				if (!IGNORE_METHOD_REGEX.matcher(method.getName()).matches() && shouldAddMethodToApi(method)) {
-					extractAndAppendResourceInfo(method, javaDoc.getJavaDoc(method), parentResource);
+					extractAndAppendResourceInfo(clazz, method, javaDoc.getJavaDoc(method), parentResource);
 				}
 			}
 		} catch (NoClassDefFoundError nEx) {
@@ -275,7 +275,7 @@ public abstract class ResourceParser {
 	 * @param method The Method to inspect
 	 * @return The Verb and Name (partial url if a resource needs to be created) of this method
 	 */
-	protected abstract Map<RamlActionType, String> getHttpMethodAndName(Method method);
+	protected abstract Map<RamlActionType, String> getHttpMethodAndName(Class<?> clazz, Method method);
 
 	/**
 	 * Extracts relevant info from a Java method and converts it into a RAML resource
@@ -284,7 +284,7 @@ public abstract class ResourceParser {
 	 * @param docEntry The associated JavaDoc (may be null)
 	 * @param parentResource The Resource which contains this method
 	 */
-	protected abstract void extractAndAppendResourceInfo(Method method, JavaDocEntry docEntry, RamlResource parentResource);
+	protected abstract void extractAndAppendResourceInfo(Class<?> clazz, Method method, JavaDocEntry docEntry, RamlResource parentResource);
 
 	/**
 	 * Checks is this api call is made directly on a resource without a trailing command in the URL. eg: POST on
diff --git a/springmvc-raml-parser/src/main/java/com/phoenixnap/oss/ramlapisync/parser/SpringMvcResourceParser.java b/springmvc-raml-parser/src/main/java/com/phoenixnap/oss/ramlapisync/parser/SpringMvcResourceParser.java
index bdfb498..4b92755 100644
--- a/springmvc-raml-parser/src/main/java/com/phoenixnap/oss/ramlapisync/parser/SpringMvcResourceParser.java
+++ b/springmvc-raml-parser/src/main/java/com/phoenixnap/oss/ramlapisync/parser/SpringMvcResourceParser.java
@@ -25,37 +25,20 @@ import com.phoenixnap.oss.ramlapisync.naming.NamingHelper;
 import com.phoenixnap.oss.ramlapisync.naming.Pair;
 import com.phoenixnap.oss.ramlapisync.naming.RamlHelper;
 import com.phoenixnap.oss.ramlapisync.naming.SchemaHelper;
-import com.phoenixnap.oss.ramlapisync.raml.RamlAction;
-import com.phoenixnap.oss.ramlapisync.raml.RamlActionType;
-import com.phoenixnap.oss.ramlapisync.raml.RamlMimeType;
-import com.phoenixnap.oss.ramlapisync.raml.RamlModelFactory;
-import com.phoenixnap.oss.ramlapisync.raml.RamlModelFactoryOfFactories;
-import com.phoenixnap.oss.ramlapisync.raml.RamlParamType;
-import com.phoenixnap.oss.ramlapisync.raml.RamlResource;
-import com.phoenixnap.oss.ramlapisync.raml.RamlResponse;
-import com.phoenixnap.oss.ramlapisync.raml.RamlUriParameter;
+import com.phoenixnap.oss.ramlapisync.raml.*;
 import org.slf4j.Logger;
 import org.slf4j.LoggerFactory;
 import org.springframework.stereotype.Controller;
 import org.springframework.util.StringUtils;
-import org.springframework.web.bind.annotation.PathVariable;
-import org.springframework.web.bind.annotation.RequestBody;
-import org.springframework.web.bind.annotation.RequestMapping;
-import org.springframework.web.bind.annotation.RequestMethod;
-import org.springframework.web.bind.annotation.RequestParam;
-import org.springframework.web.bind.annotation.RestController;
+import org.springframework.web.bind.annotation.*;
 
 import java.io.File;
 import java.lang.annotation.Annotation;
+import java.lang.reflect.AnnotatedElement;
 import java.lang.reflect.Method;
 import java.lang.reflect.Parameter;
-import java.util.ArrayList;
-import java.util.Collections;
-import java.util.HashMap;
-import java.util.List;
-import java.util.Map;
+import java.util.*;
 import java.util.Map.Entry;
-import java.util.TreeMap;
 
 /**
  * Service scanner that handles generation from a Spring MVC codebase
@@ -344,10 +327,10 @@ public class SpringMvcResourceParser extends ResourceParser {
 	}
 
 	@Override
-	protected Map<RamlActionType, String> getHttpMethodAndName(Method method) {
+	protected Map<RamlActionType, String> getHttpMethodAndName(Class<?> clazz, Method method) {
 		RequestMapping methodMapping = getRequestMapping(method);
-		RequestMapping classMapping = getAnnotation(method.getDeclaringClass(), RequestMapping.class, false);
-		RestController classRestController = getAnnotation(method.getDeclaringClass(), RestController.class,
+		RequestMapping classMapping = getAnnotation(clazz, RequestMapping.class, false);
+		RestController classRestController = getAnnotation(clazz, RestController.class,
 				false);
 		Controller classController = getAnnotation(method.getDeclaringClass(), Controller.class, false);
 
@@ -427,11 +410,11 @@ public class SpringMvcResourceParser extends ResourceParser {
 	}
 
 	@Override
-	protected void extractAndAppendResourceInfo(Method method, JavaDocEntry docEntry, RamlResource parentResource) {
+	protected void extractAndAppendResourceInfo(Class<?> clazz, Method method, JavaDocEntry docEntry, RamlResource parentResource) {
 
 		RamlModelFactory ramlModelFactory = RamlModelFactoryOfFactories.createRamlModelFactory();
 
-		Map<RamlActionType, String> methodActions = getHttpMethodAndName(method);
+		Map<RamlActionType, String> methodActions = getHttpMethodAndName(clazz, method);
 		for (Entry<RamlActionType, String> methodAction : methodActions.entrySet()) {
 			RamlAction action = ramlModelFactory.createRamlAction();
 			RamlActionType apiAction = methodAction.getKey();

```

**Comment by kurtpa** [comment]
released with 0.8.7 thanks bendem for the merge and stojsavljevi for applying it

**Comment by bendem** [comment]
Awesome, thanks
